apps/balancer-tools/src/app/poolsimulator/(components)/SelectPoolType.tsx

```diff
@@ -1,17 +1,19 @@
 import { Close, DialogClose } from "@radix-ui/react-dialog";
 
-import { PoolType, usePoolSimulator } from "#/contexts/PoolSimulatorContext";
+import { usePoolFormContext } from "#/contexts/FormContext";
+import { PoolType } from "#/contexts/PoolSimulatorContext";
 
-export function SelectPoolType({
-  selectedType,
-}: {
-  selectedType: PoolType;
-  setSelectedType: (value: PoolType) => void;
-}) {
-  const { setPoolType } = usePoolSimulator();
+export function SelectPoolType({ selectedType }: { selectedType: PoolType }) {
+  const { setData, data } = usePoolFormContext();
+  const event = new CustomEvent("changePoolType");
```

Comment:
can you elaborate more about why you chose to create a custom event here?

Reply:
This is needed to reset the form when the user changes the poolType.

The other solution I tried was to use an `useEffect` on the `data.poolType` on the forms component. However, when we changed from initial to custom data this `useEffect` was being triggered leading to form resetting when the user navigated with the tabs.

So this event is to be watched on the form component to reset the fields. I also thought about using a flag state to manage this, but the event seems more appropriate. What is your opinion about it?